Microsoft.Toolkit.Uwp.UI.Controls/FlexPanel/FlexPanel.cs

```diff
@@ -0,0 +1,1576 @@
+            "FlexItem",
+            typeof(object),
+            typeof(FlexPanel),
+            new PropertyMetadata(null));
+
+        private static FlexItem GetFlexItem(UIElement element)
+        {
+            if (element is null)
+            {
+                return null;
+            }
+
+            if (element is FlexPanel flexPanel)
+            {
+                return flexPanel.root;
+            }
+
+            FlexItem item = null;
+            try
+            {
+                item = (FlexItem)element.GetValue(FlexItemProperty);
+            }
+            catch (Exception)
+            {
+            }
+
+            if (item is null)
+            {
+                item = new FlexItem();
+                element.SetValue(FlexItemProperty, item);
+            }
+
+            return item;
+        }
+
+        private static void SetFlexItem(UIElement element, FlexItem value)
+        {
+            element?.SetValue(FlexItemProperty, value);
+            UpdateItemProperties(element, value);
+        }
+
+        private static void InternalInvalidateArrange(UIElement element)
+        {
+            if (element is FlexPanel)
+            {
+                element.InvalidateArrange();
+            }
+            else if (element is FrameworkElement frameworkElement && frameworkElement.Parent is FlexPanel flexPanel)
+            {
+                flexPanel.InvalidateArrange();
+            }
+        }
+
+        private static void UpdateItemProperties(UIElement view, FlexItem item)
+        {
+            item.IsVisible = view.Visibility == Visibility.Visible;
+
+            if (view is FrameworkElement element)
+            {
+                item.MarginLeft = (double)element.Margin.Left;
+                item.MarginTop = (double)element.Margin.Top;
+                item.MarginRight = (double)element.Margin.Right;
+                item.MarginBottom = (double)element.Margin.Bottom;
+            }
+
+            if (view is Control control)
+            {
+                item.SetPadding(control.Padding);
+            }
+            else if (view is Border border)
+            {
+                item.SetPadding(border.Padding);
+            }
+            else if (view is TextBlock textBlock)
+            {
+                item.SetPadding(textBlock.Padding);
+            }
+#if __ANDROID__ || __WASM__
+#else
+            else if (view is RichTextBlock richTextBlock)
+            {
+                item.SetPadding(richTextBlock.Padding);
+            }
+#endif
+        }
+
+        private readonly FlexItem root = new FlexItem();
+
+        private FlexItem AddChild(FrameworkElement view)
+        {
+            if (root == null)
+            {
+                return null;
+            }
+
+            view.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            var item = (view as FlexPanel)?.root ?? new FlexItem();
```

Comment:
Agree with @skendrot, you're trying to abuse `Panel` for what should be an `ItemsControl` then it automatically controls creating and managing these wrapping 'FlexItem`s for you.

You either need to just use the Attached Properties directly in the panel with the given items within the layout pass OR you should make it an ItemsControl which will containerize everything in it and then you can just have the properties on the FlexItem directly. Due to the number of properties and complexity here, the latter of using ItemsControl as a base seems like the better option.

The only problem is then it's hard to add selection later unless we use ListViewBase instead... thoughts @skendrot?